enh(doc) missing monitoring proc december Pack release (#1195)
centreon/centreon-documentation: en/integrations/plugin-packs/procedures/cloud-azure-storage-acr.md

---
id: cloud-azure-storage-acr
title: Azure Container Registry
---

## Overview

Azure Container Registry is a managed, private Docker registry service based on the open-source Docker Registry 2.0. Create and maintain Azure
Container Registries to store and manage your private Docker container images and related artifacts.

The Centreon Plugin *Azure Container Registry* can rely on Azure API or Azure CLI to collect the metrics related to the *ACR* service.

## Pack Assets

### Monitored Objects

Available indicators and metrics:
* Azure Container Registries
     * Storage-usage

### Discovery rules

> The Centreon Pack *Azure Container Registry* includes a Host Discovery *provider* to automatically discover the Azure instances of a given

Reviewer: ```suggestion
The Centreon Pack *Azure Container Registry* includes a Host Discovery provider to automatically discover the Azure instances of a given
```

subscription and add them to the Centreon configuration.
This provider is named **Microsoft Azure Container Registry**:

![image](../../../assets/integrations/plugin-packs/procedures/cloud-azure-storage-acr-provider.png)

> This discovery feature is only compatible with the 'api' custom mode. 'azcli' is not supported.
More information about the Host Discovery module is available in the Centreon documentation:
[Host Discovery](../../../monitoring/discovery/hosts-discovery.html)

### Collected metrics

<!--DOCUSAURUS_CODE_TABS-->

<!--Storage-usage-->

| Metric Name                                            | Description  | Unit |
|:-------------------------------------------------------|:-------------|:-----|
| *instance*#azure.containerregistry.storage.used.bytes  | Storage used | B    |

<!--END_DOCUSAURUS_CODE_TABS-->

## Prerequisites

> To get data from Azure Services, following methods are available:

Reviewer: ```suggestion
To get data from Azure Services, the following methods are available:
```

    * Azure API ('api') 
    * Azure CLI ('azcli')

Centreon recommends to use the API instead of the CLI for the following reasons:

>     * API is much more efficient by avoiding CLI binary execution

Reviewer: ```suggestion
    * the API is much more efficient because it avoids CLI binary execution
```

>     * API supports application authentication while CLI does not (yet)

Reviewer: ```suggestion
    * the API supports application authentication while CLI does not (yet)
```

<!--DOCUSAURUS_CODE_TABS-->

<!--Azure Monitor API-->

To use the 'api' custom mode, make sure to obtain the required information using the 
how-to below. Keep it safe until including it in a Host or Host Template definition.

* Create an *application* in Azure Active Directory:
    - Log in to your Azure account.
    - Select *Azure Active directory* in the left sidebar.
    - Click on *App registrations*.
    - Click on *+ Add*.

>     - Enter Centreon as the application name (or any name of your choice), select application type(api) and sign-on-url.

Reviewer: ```suggestion
    - Enter Centreon as the application name (or any name of your choice), select application type (api) and sign-on-url.
```

    - Click on the *Create* button.

* Get *Subscription ID*
    - Log in to your Azure account.
    - Select *Subscriptions* in the left sidebar.
    - Select whichever subscription is needed.
    - Click on *Overview*.

>     - **Copy the Subscription ID.**

Reviewer: ```suggestion
    - Copy the Subscription ID.
```

* Get *Tenant ID*
    - Log in to your Azure account.
    - Select *Azure Active directory* in the left sidebar.
    - Click on *Properties*.

>     - **Copy the directory ID.**

Reviewer: ```suggestion
    - Copy the directory ID.
```

* Get *Client ID*
    - Log in to your Azure account.
    - Select *Azure Active directory* in the left sidebar.
    - Click on *Enterprise applications*.
    - Click on *All applications*.
    - Select the application previously created.
    - Click on *Properties*.

>     - **Copy the Application ID.**

Reviewer: ```suggestion
    - Copy the Application ID.
```

* Get *Client secret*
    - Log in to your Azure account.
    - Select *Azure Active directory* in the left sidebar.
    - Click on *App registrations*.
    - Select the application previously created.
    - Click on *All settings*.
    - Click on *Keys*.
    - Enter the key description and select the duration.
    - Click on *Save*.

>     - **Copy and store the key value. You won't be able to retrieve it after you leave this page.**

Reviewer: ```suggestion
    - Copy and store the key value. **You won't be able to retrieve it after you leave this page.**
```

> Please make sure to **assign the Monitoring Reader** role to the application.

Reviewer: ```suggestion
Please make sure to assign the **Monitoring Reader** role to the application.
```

<!--Azure AZ CLI-->

To use the 'azcli' custom mode, install the required packages on every Centreon poller expected to 
monitor Azure Resources using CLI:

- The CLI needs at least Python version 2.7
(<https://github.com/Azure/azure-cli/blob/dev/doc/install_linux_prerequisites.md>).

On RPM-Based distributions, use the command below to install it using *root* or 'sudo':

```shell
sudo rpm --import https://packages.microsoft.com/keys/microsoft.asc
sudo echo -e "[azure-cli]\nname=Azure CLI\nbaseurl=https://packages.microsoft.com/yumrepos/azure-cli\nenabled=1\ngpgcheck=1\ngpgkey=https://packages.microsoft.com/keys/microsoft.asc" > /etc/yum.repos.d/azure-cli.repo
sudo yum install azure-cli
```

> Then, use the *centreon-engine* account to obtain a token using command below: 

Reviewer: ```suggestion
Then, use the **centreon-engine** account to obtain a token using the command below: 
```

```shell
su - centreon-engine
az login
```

The shell will output this message including an authentication code:

    *To sign in, use a web browser to open the page https://microsoft.com/devicelogin* 
    *and enter the code CWT4WQZAD to authenticate.*

Go to <https://microsoft.com/devicelogin> and enter the code. 

> Connect using a monitoring service account, as a result, the shell should prompt

Reviewer: ```suggestion
Connect using a monitoring service account. As a result, the shell should give you
```

> information below:

Reviewer: ```suggestion
the following information:
```

```shell
    [
      {
        "cloudName": "AzureCloud",
        "id": "0ef83f3a-d83e-2039-d930-309df93acd93d",
        "isDefault": true,
        "name": "N/A(tenant level account)",
        "state": "Enabled",
        "tenantId": "0ef83f3a-03cd-2039-d930-90fd39ecd048",
        "user": {
          "name": "email@mycompany.onmicrosoft.com",
          "type": "user"
        }
      }
    ]
```

> Credentials are now stored locally in the .accessTokens.json file so the Plugin 

Reviewer: ```suggestion
Credentials are now stored locally in the **.accessTokens.json** file so the Plugin 
```

can use it. 

<!--END_DOCUSAURUS_CODE_TABS-->

## Setup 

<!--DOCUSAURUS_CODE_TABS-->

<!--Online IMP Licence & IT-100 Editions-->

1. Install the Centreon Plugin package on every Centreon poller expected to monitor *Azure Container Registry* resources:

```bash
yum install centreon-plugin-Cloud-Azure-Storage-ContainerRegistry-Api
```

> 2. On the Centreon Web interface, install the *Azure Container Registry* Pack on the **Configuration > Plugin Packs > Manager** page

Reviewer: ```suggestion
2. On the Centreon Web interface, install the *Azure Container Registry* Pack on the **Configuration > Plugin Packs > Manager** page.
```

| X         | AZURETENANT        | Tenant ID                                          |
| X         | AZURECLIENTID      | Client ID                                          |
| X         | AZURECLIENTSECRET  | Client secret                                      |
| X         | AZURERESOURCE      | ID or name of the Container Registry resource      |
|           | AZURERESOURCEGROUP | Associated Resource Group if resource name is used |

<!--Azure AZ CLI-->

| Mandatory | Nom                | Description                                        |
|:----------|:-------------------|:---------------------------------------------------|
| X         | AZURECUSTOMMODE    | Custom mode 'azcli'                                |
| X         | AZURESUBSCRIPTION  | Subscription ID                                    |
| X         | AZURERESOURCE      | ID or name of the Container Registry resource      |
|           | AZURERESOURCEGROUP | Associated Resource Group if resource name is used |

<!--END_DOCUSAURUS_CODE_TABS-->

## How to check in the CLI that the configuration is OK and what are the main options for ?

> Once the Plugin installed, log into your Centreon Poller CLI using the *centreon-engine* 

Reviewer: ```suggestion
Once the Plugin is installed, log into your Centreon Poller CLI using the **centreon-engine** 
```
